src/Microsoft.MobileBlazorBindings.Hosting/IWebViewIPCAdapter.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.MobileBlazorBindings.Hosting
+{
+    public interface IWebViewIPCAdapter
+    {
+        EventHandler<string> OnWebMessageReceived { get; set; }
+        void SendMessage(string message);
+        void BeginInvoke(Action action);
+
+
+        // [JSInvokable] calls
+        Task DispatchEvent(WebEventDescriptor eventDescriptor, string eventArgsJson);
+#pragma warning disable CA1054 // URI-like parameters should not be strings
```

Comment:
I can see quite a lot of such suppressions. Would it make sense to disable this diagnostic in editorconfig?

Reply:
That's a fair question and I've gone back and forth on this. Much of this code will receive a large rewrite when it gets productized so _for the most part_ I tried to localized the suppressions.